code/builders/builder-vite/src/build.ts

```diff
@@ -28,5 +35,18 @@ export async function build(options: Options) {
   }).build;
 
   const finalConfig = await presets.apply('viteFinal', config, options);
+
+  const turbosnapPlugin = findPlugin(finalConfig, 'rollup-plugin-turbosnap');
+  if (turbosnapPlugin) {
+    logger.warn(`Found 'rollup-plugin-turbosnap' which is now included by default in Storybook 8.`);
+    logger.warn(
+      `Removing from your plugins list. Ensure you pass \`--webpack-stats-json\` to generate stats.`
+    );
+    finalConfig.plugins = finalConfig.plugins?.filter((p) => p !== turbosnapPlugin);
```

Comment:
We already have utility functions to detect and remove plugins, it would probably be best to reuse those.

See https://github.com/storybookjs/storybook/blob/next/code/frameworks/svelte-vite/src/utils.ts#L20-L26 for detecting plugins and https://github.com/storybookjs/storybook/blob/next/code/frameworks/sveltekit/src/preset.ts#L29-L32 for removing plugins